server, ui: fix data distribution endpoint

vilterp opened the pull request:
Still needs some design tweaks (@couchand suggested greying out dropped tables instead of red-asterisk-ing them), but the backend part is ready for review.

Fixes #24853

![image](https://user-images.githubusercontent.com/7341/46758819-cdd3b100-cc9b-11e8-9a38-e134ec0ec3bb.png)

Commit:
server: fix crash of data distribution endpoint after dropping a table

The crash was caused by trying to get the zone config with `SHOW ZONE
CONFIGURATION FOR TABLE` for a table that had been dropped, producing an
unexpected error.

Release note: None

Commit:
server: fix benchmark of data distribution endpoint

Release note: None

Commit:
ui: data distribution: grey out dropped tables

Also add a blurb under the zone config list explaining what greying out
means.

Release note: None

Commit:
ui: data distribution: tweak zone config styling

Just show the SQL, not the cli specifier, since we don't use CLI
specifiers anymore.

Release note: None

Commit:
ui: refactor: provide default className for Loading component

Most usages pass `loading-image loading-image__spinner-left` to the
className prop; make that the default and remove the prop for those
usages.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/31117)
<!-- Reviewable:end -->

vilterp commented:
Roachtest timed out in CI — I think it's spurious; hit rebuild.

couchand commented on pkg/ui/src/views/cluster/containers/dataDistribution/replicaMatrix.tsx:
```diff
@@ -94,6 +100,32 @@ class ReplicaMatrix extends Component<ReplicaMatrixProps, ReplicaMatrixState> {
+  rowLabel(row: FlattenedNode<SchemaObject>) {
+    const text = this.rowLabelText(row);
+
+    const droppedIndicator = row.data.droppedAt
```
Since you've got the conditional below anyway, you can get rid of this one entirely.

vilterp replied:
Got rid of this anyway since changed from asterisk to greying out dropped tables

couchand commented on pkg/ui/src/views/shared/components/loading/index.tsx:
```diff
@@ -12,11 +12,12 @@ interface LoadingProps {
 // * loading prop is true.
 // *
 export default function Loading(props: LoadingProps) {
+  const className = props.className || "loading-image loading-image__spinner-left";
```
Nice.  Heck, why not `const imagePath = props.image || spinner;`, too?

vilterp replied:
Done. Hilarious how many call sites pass the same thing, and that we didn't do this earlier 😂

couchand reviewed (approved):
Looks great!

Commit:
ui: refactor: make spinner default image to Loading component

...and remove all that prop from all the usages that passed it.

Release note: None

vilterp commented:
bors r+
